lib/msf/core/session/provider/single_command_shell.rb

```diff
@@ -119,7 +119,7 @@ def set_is_echo_shell(timeout, command_separator)
     cmd = "echo #{numeric_token}"
     shell_write(cmd + "#{command_separator}echo #{token}#{command_termination}")
     res = shell_read_until_token(token, 0, timeout)
-    @is_echo_shell = res.include?(cmd)
+    @is_echo_shell = res ? res.include?(cmd) : false
```

Comment:
```suggestion
    @is_echo_shell = res&.include?(cmd)
```

Reply:
Just a heads up that these would be semantically different

Before

```
3.2.5 :001 > res = nil
3.2.5 :002 > @is_echo_shell = res&.include?(cmd)
3.2.5 :003 > @is_echo_shell = res ? res.include?(cmd) : false
 => false 
```

After

```
3.2.5 :001 > res = nil
3.2.5 :002 > @is_echo_shell = res&.include?(cmd)
3.2.5 :003 > puts @is_echo_shell.inspect
nil
 => nil 
```

Not a blocker; `!!res&.include?(cmd)` would be closer to the existing semantics

Reply:
Thanks!

Reply:
Yeah, looks like if it ends up as `nil`, it'll just keep trying this every time; which while probably harmless, is probably not intended. Going with `false` seems better.